compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -306,7 +306,15 @@ fn encode_substs<'tcx>(
                     s.push_str(&encode_region(tcx, region, dict, options));
                 }
                 GenericArgKind::Type(ty) => {
-                    s.push_str(&encode_ty(tcx, ty, dict, options));
+                    match *ty.kind() {
```

Comment:
Why not just change the way we encode `ty::Generator` so that we never encounter `GeneratorWitness` when encoding types?

Reply:
Do you have an example of what you mean? I don't think we don't want to encounter a ty::GeneratorWitness but want to encode it correctly as part of substs as specified by the encoding.

Reply:
I would walk up the call tree and modify the code that's responsible for encoding generators to handle the generator witness there (see https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/list/struct.List.html#method.as_generator) rather than doing it here.

Reply:
Done.

Reply:
Are you suggesting to encode `ty::GeneratorWitness` in `encode_ty`? The reason why it's encoded in `encode_substs` and not in `encode_ty` is because it should never occur as a standalone type in `encode_ty`, and if so, it should trigger an unexpected ty bug.

Reply:
After chatting with @bjorn3 and @bjorn3 explaining me the `ty::GeneratorWitness` in more detail, we decided that we don't want to encode `ty::GeneratorWitness`. So I just changed these changes to expect but not encode it.